packages/language-server/test/plugins/typescript/features/HoverProvider.test.ts

```diff
@@ -6,184 +6,211 @@ import { Document, DocumentManager } from '../../../../src/lib/documents';
 import { LSConfigManager } from '../../../../src/ls-config';
 import { HoverProviderImpl } from '../../../../src/plugins/typescript/features/HoverProvider';
 import { LSAndTSDocResolver } from '../../../../src/plugins/typescript/LSAndTSDocResolver';
+import { __resetCache } from '../../../../src/plugins/typescript/service';
 import { pathToUrl } from '../../../../src/utils';
 
 const testDir = path.join(__dirname, '..');
 
-describe('HoverProvider', () => {
-    function getFullPath(filename: string) {
-        return path.join(testDir, 'testfiles', 'hover', filename);
-    }
-
-    function setup(filename: string) {
-        const docManager = new DocumentManager(
-            (textDocument) => new Document(textDocument.uri, textDocument.text)
-        );
-        const lsAndTsDocResolver = new LSAndTSDocResolver(
-            docManager,
-            [testDir],
-            new LSConfigManager()
-        );
-        const provider = new HoverProviderImpl(lsAndTsDocResolver);
-        const document = openDoc(filename);
-        return { provider, document };
-
-        function openDoc(filename: string) {
-            const filePath = getFullPath(filename);
-            const doc = docManager.openDocument(<any>{
-                uri: pathToUrl(filePath),
-                text: ts.sys.readFile(filePath) || ''
-            });
-            return doc;
+function test(useNewTransformation: boolean) {
```

Comment:
The tests are mainly changed like this: Extract exising tests into a function which accepts `useNewTransformation` as a parameter and passes that to the `LSConfigManager`. There are some small adjustments in a few tests but mostly they run without any difference for both new and old transformations.